formats/ome-converter-plugin/tests/test_main.py

```diff
@@ -0,0 +1,109 @@
+"""Testing of Ome Converter."""
+import os
+import pathlib
+import sys
+from multiprocessing import cpu_count
+
+import filepattern as fp
+import numpy as np
+import requests
+from bfio import BioReader, BioWriter
+from PIL import Image
+from skimage import io
+
+from polus.plugins.formats.ome_converter.image_converter import image_converter
+
+dirpath = os.path.abspath(os.path.join(__file__, "../.."))
+sys.path.append(dirpath)
+
+inpDir = pathlib.Path(dirpath, "data/input")
+outDir = pathlib.Path(dirpath, "data/out")
+omeDir = pathlib.Path(dirpath, "data/outome")
+zarrDir = pathlib.Path(dirpath, "data/outzarr")
+if not inpDir.exists():
+    inpDir.mkdir(parents=True, exist_ok=True)
+if not outDir.exists():
+    outDir.mkdir(exist_ok=True, parents=True)
+if not omeDir.exists():
+    omeDir.mkdir(exist_ok=True, parents=True)
+if not zarrDir.exists():
+    zarrDir.mkdir(parents=True, exist_ok=True)
+
+imagelist = {
+    "0.tif": "https://osf.io/j6aer/download",
+    "img_r001_c001.ome.tif": "https://github.com/usnistgov/WIPP/raw/master/data/PyramidBuilding/inputCollection/img_r001_c001.ome.tif",
+    "00001_01.ome.tiff": "https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/MitoCheck/00001_01.ome.tiff",
+}
+pattern = ".*"
+
+
+def download_images():
+    """Download test /Users/abbasih2/Documents/Polus_Repos/polus-plugins/formats/ome-converter-plugin/data/inputimages."""
+    for file, url in imagelist.items():
+        r = requests.get(url)
+        with open(inpDir.joinpath(file), "wb") as fw:
+            fw.write(r.content)
+
+
+def generate_sythetic_images():
+    """Generate synthetic images."""
+    arr = np.arange(0, 1048576, 1, np.uint8)
+    arr = np.reshape(arr, (1024, 1024))
+    image = Image.fromarray(arr)
+    image.save(inpDir.joinpath("syn_image_1.png"))
+
+
+def omewrite(image, outfile):
+    """Write images to using bfiowriter."""
+    with BioWriter(file_path=outfile) as bw:
+        bw.X = image.shape[0]
+        bw.Y = image.shape[1]
+        bw.dtype = image.dtype
+        bw[:] = image
+
+
+def test_image_converter_omezarr():
```

Comment:
Once the above functions and variables are converted into fixtures, this test and the other tests should all be turned into parametrized tests.

Reply:
fixed